src/pages/filter/ui/enum-panel.tsx

```diff
@@ -20,124 +21,162 @@ export const EnumPanel = observer(
+    const filteredVariants = variants.filter((variant: any[]) =>
+      variant[0]
+        .toLocaleLowerCase()
+        .startsWith(searchValue.toLocaleLowerCase()),
+    )
 
-    const handleSelect = (index: number, checked: boolean) => {
-      const localVariantList = [...variantList]
+    const groupsPerPage = 12
+    const chunks = createChunks(filteredVariants, groupsPerPage)
 
-      const currentIndex = currentPage * groupsPerPage + index
+    const handleCheckGroupItem = (
+      checked: boolean,
+      variant: [string, number],
+    ) => {
+      if (checked) {
+        const localVariantNameList = [...selectedVariants, variant]
 
-      localVariantList[currentIndex].isChecked = checked
+        setSelectedVariants(localVariantNameList)
+      } else {
+        const localVariantNameList = selectedVariants.filter(
+          element => element[0] !== variant[0],
+        )
 
-      setVariantList(localVariantList)
+        setSelectedVariants(localVariantNameList)
+      }
     }
 
-    const stringedVariants = JSON.stringify(variants)
+    const group = filterStore.selectedGroupItem.vgroup
+    const groupItemName: string = filterStore.selectedGroupItem.name
 
     useEffect(() => {
-      setVariantList(variants)
-
-      // eslint-disable-next-line react-hooks/exhaustive-deps
-    }, [stringedVariants])
-
-    const updateSubattributes = (
-      shouldAddSubattributes?: boolean,
-    ): string[] => {
-      const checkedVariantNames: string[] = []
-
-      variantList.forEach(element => {
-        const { variant, isChecked } = element
-
-        if (shouldAddSubattributes && isChecked) {
-          filterStore.addSelectedFilters({
-            group: filterStore.selectedGroupItem.vgroup,
-            groupItemName: filterStore.selectedGroupItem.name,
-            variant,
-          })
-
-          checkedVariantNames.push(variant[0])
-        } else {
-          const variantQuantity =
-            filterStore.selectedFilters?.[`${vgroup}`]?.[`${groupItemName}`]?.[
-              `${variant[0]}`
-            ]
-
-          const isVariantChecked = Boolean(variantQuantity)
-
-          if (!isVariantChecked) return
-
-          filterStore.removeSelectedFilters({
-            group: filterStore.selectedGroupItem.vgroup,
-            groupItemName: filterStore.selectedGroupItem.name,
-            variant,
-          })
-
-          datasetStore.removeCondition(
-            {
-              subGroup: filterStore.selectedGroupItem.name,
-              itemName: variant[0],
-            },
-            false,
-          )
-        }
-      })
+      setSearchValue('')
+      setCurrentPage(0)
+    }, [groupItemName])
 
-      return checkedVariantNames
-    }
+    const [shouldClear, setShouldClear] = useState(false)
 
     const handleClear = () => {
-      updateSubattributes()
-      datasetStore.fetchDsStatAsync()
+      const localSelectedFilters = JSON.parse(
+        JSON.stringify(filterStore.selectedFilters),
+      )
```

Comment:
better to use cloneDeep from `'lodash/cloneDeep'`. It's faster and keeps the original type. While JSON.parse/stringify always returns `any`